src/main/java/com/commercetools/config/ctpTypes/ExpectedCtpTypes.java

```diff
@@ -0,0 +1,79 @@
+package com.commercetools.config.ctpTypes;
+
+import io.sphere.sdk.types.Type;
+
+import javax.annotation.Nonnull;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import static io.sphere.sdk.json.SphereJsonUtils.readObjectFromResource;
+import static java.util.Collections.unmodifiableSet;
+import static java.util.stream.Collectors.toSet;
+
+/**
+ * Reading directory content from packed resources is a sophisticated task, thus we define the expected CTP {@link Type}
```

Comment:
> Why is it sophisticated?

because i have to unzip currently executing the jar itself using libraries  to access compressed resources content.

> I thought you're also reading the .json files from ctp/types/ to create new type?

read resource file with known full path is a native java feature, reading dir isn't

https://stackoverflow.com/questions/1429172/how-do-i-list-the-files-inside-a-jar-file
https://stackoverflow.com/questions/11012819/how-can-i-get-a-resource-folder-from-inside-my-jar-file